src/symbolic_dimensions.jl

```diff
@@ -35,49 +34,54 @@ to one which uses `Dimensions` as its dimensions (i.e., base SI units)
 `expand_units`.
 """
 struct SymbolicDimensions{R} <: AbstractDimensions{R}
-    _data::SA.SparseVector{R}
-
-    SymbolicDimensions(data::SA.SparseVector) = new{eltype(data)}(data)
-    SymbolicDimensions{_R}(data::SA.SparseVector) where {_R} = new{_R}(data)
+    nzdims::Vector{UInt8}
+    nzvals::Vector{R}
```

Comment:
Could these be `_nzdims` and `_nzvals` to indicate they should not be accessed by the user? Since all other `AbstractDimensions` objects typically have each field corresponding to a specific physical dimension.

Reply:
I was considering this as well 😅 It didn't seem necessary to me though since you can't access them accidentally anyway: You have to use `getfield(dims, :nzvals)` etc. since `dims.nzvals` will throw an error (there's no symbolic dimension `:nzvals`). So using `_nzvals` seemed mainly to increase the number of characters in the implementation.

What do you think?

Reply:
Do you know if hitting the tab key on `dim.` in the REPL will print out the properties, or the fields? (Sent from mobile so I can’t check right now)

Reply:
It depends on whether you define `Base.propertynames` (as done in this PR) or not (as currently done on the master branch):

In this PR:
```julia
julia> using DynamicQuantities

julia> dims = SymbolicDimensions(; m = 3, s = -1)
m³ s⁻¹

julia> dims.
A         C         F         G         GHz       GV        GW        GeV       Gm        Gpc       Gyr       Hz        J         K         L         L_bol0    L_sun     MHz
MV        MW        M_earth   M_jup     M_sun     MeV       Mm        Mpc       Myr       N         N_A       Pa        R         R_earth   R_jup     R_sun     Ryd       T
TeV       V         W         a_0       alpha     atm       au        bar       c         cd        cm        dL        day       dm        e         eV        eps_0     fm
fs        g         h         hbar      hr        kA        kHz       kJ        kPa       kV        kW        k_B       k_e       keV       kg        km        kpc       kyr
ly        m         mA        mK        mL        mV        m_e       m_n       m_p       mcd       meV       mg        min       mm        mmol      mohm      mol       ms
mu_0      mΩ        nA        nm        ns        ohm       pc        pm        ps        s         sigma_T   sigma_sb  u         uA        ug        um        us        yr
Ω         μA        μg        μm        μs
```

On the master branch:
```julia
julia> using DynamicQuantities

julia> dims = SymbolicDimensions(; m = 3, s = -1)
m³ s⁻¹

julia> dims._data # (Autocompleted to _data after pressing tab)
```

Reply:
Cool! Okay I don’t think there’s a need for this change then. Unless we want to indicate it’s a private field at the source code level.